File: proactive-frontend/src/app/models/project.ts
export type ProjectStatus   = 'PLANNING' | 'ACTIVE' | 'ON_HOLD' | 'COMPLETED' | 'CANCELLED';
export type ProjectPriority = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface Project {
  id:           number;
  name:         string;
  description:  string | null;
  startDate:    string | null;
  endDate:      string | null;
  status:       ProjectStatus;
  priority:     ProjectPriority;
  budget:       number | null;
  progressPct:  number;
  createdAt:    string;
  updatedAt:    string;
  managerId:    number | null;
  managerName:  string | null;
  managerEmail: string | null;
  createdById:  number | null;
  createdByName:string | null;
}

export interface ProjectRequest {
  name?:        string;
  description?: string;
  startDate?:   string;
  endDate?:     string;
  status?:      ProjectStatus;
  priority?:    ProjectPriority;
  budget?:      number;
  progressPct?: number;
  managerId?:   number;
}

export interface ProjectStats {
  total:     number;
  planning:  number;
  active:    number;
  on_hold:   number;
  completed: number;
  cancelled: number;
}

export const STATUS_LABELS: Record<ProjectStatus, string> = {
  PLANNING:  'Planification',
  ACTIVE:    'En cours',
  ON_HOLD:   'En pause',
  COMPLETED: 'Terminé',
  CANCELLED: 'Annulé',
};

export const PRIORITY_LABELS: Record<ProjectPriority, string> = {
  LOW:      'Faible',
  MEDIUM:   'Moyenne',
  HIGH:     'Haute',
  CRITICAL: 'Critique',
};

export const STATUS_COLORS: Record<ProjectStatus, string> = {
  PLANNING:  '#f0a500',
  ACTIVE:    '#1a73e8',
  ON_HOLD:   '#8090a8',
  COMPLETED: '#27ae60',
  CANCELLED: '#e74c3c',
};

export const PRIORITY_COLORS: Record<ProjectPriority, string> = {
  LOW:      '#27ae60',
  MEDIUM:   '#f0a500',
  HIGH:     '#e67e22',
  CRITICAL: '#e74c3c',
};

export const ALL_STATUSES:   ProjectStatus[]   = ['PLANNING','ACTIVE','ON_HOLD','COMPLETED','CANCELLED'];
export const ALL_PRIORITIES: ProjectPriority[] = ['LOW','MEDIUM','HIGH','CRITICAL'];

File: demo/src/main/java/com/proactive/demo/model/Project.java
package com.proactive.demo.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "projects")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status = Status.PLANNING;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private Priority priority = Priority.MEDIUM;

    /** Type de projet */
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private ProjectType type = ProjectType.OTHER;

    /** Méthodologie de travail */
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Methodology methodology = Methodology.AGILE;

    /** Outils utilisés (liste séparée par virgules : "Jira,Slack,GitHub") */
    @Column(columnDefinition = "TEXT")
    private String tools;

    /** Budget estimé */
    private Double budget;

    /** Pourcentage d'avancement (0-100) */
    @Column(name = "progress_pct")
    private Integer progressPct = 0;

    /** Manager principal du projet */
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "manager_id")
    private User manager;

    /** Créateur du projet */
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "created_by")
    private User createdBy;

    /** Membres du projet */
    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<ProjectMember> members = new ArrayList<>();

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // ── Enums ────────────────────────────────────────────────────

    public enum Status {
        PLANNING, ACTIVE, ON_HOLD, COMPLETED, CANCELLED
    }

    public enum Priority {
        LOW, MEDIUM, HIGH, CRITICAL
    }

    public enum ProjectType {
        SOFTWARE,       // Développement logiciel
        INFRASTRUCTURE, // Infrastructure / DevOps
        MARKETING,      // Campagne marketing
        DESIGN,         // Design / UX
        RESEARCH,       // Recherche & Innovation
        TRAINING,       // Formation / Education
        CONSULTING,     // Consulting
        OTHER           // Autre
    }

    public enum Methodology {
        AGILE,     // Agile / Scrum
        KANBAN,    // Kanban
        WATERFALL, // Cycle en V
        LEAN,      // Lean
        DEVOPS,    // DevOps
        PRINCE2,   // PRINCE2
        PMI,       // PMI / PMBOK
        HYBRID,    // Hybride
        OTHER      // Autre
    }
}
